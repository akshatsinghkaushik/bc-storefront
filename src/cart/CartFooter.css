.cart-footer {
  padding: 2rem 0 2rem 0;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
}
.cart-footer-button {
  margin: 0 2rem 0 2rem;
  background: none;
  background-color: black;
  color: white;
  border: none;
  padding: 1.2rem;
  font: inherit;
  cursor: pointer;
  outline: inherit;
  font-weight: 400;
  letter-spacing: 1px;
  font-size: 13px;
}

.continue-shopping {
  color: black;
  font-size: 14px;
  font-weight: 500;
  text-decoration: none;
  border-bottom: 1.5px solid #000000;
  line-height: 23px;
  transform: translateY(-1rem);
}
.continue-shopping:visited {
  color: black !important;
}
