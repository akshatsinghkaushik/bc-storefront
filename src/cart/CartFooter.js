import React, { useContext } from 'react';
import { Link } from 'react-router-dom';
import './CartFooter.css';
import CartContext from '../context/CartContext';

const CartFooter = () => {
  const cartContext = useContext(CartContext);

  return (
    <div className="cart-footer">
      <Link to={'/'} className="continue-shopping">
        CONTINUE SHOPPING
      </Link>
      <button className="cart-footer-button">
        CHECKOUT (${Number(cartContext.getCartTotal()).toFixed(2)})
      </button>
    </div>
  );
};

export default CartFooter;
