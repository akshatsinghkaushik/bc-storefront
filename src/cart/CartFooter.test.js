import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import CartFooter from './CartFooter';
import MockCartProvider from '../tests/MockCartProvider';

it('CartFooter Renders correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <MockCartProvider>
          <CartFooter />
        </MockCartProvider>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
