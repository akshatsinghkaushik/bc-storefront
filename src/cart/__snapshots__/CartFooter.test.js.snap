// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartFooter Renders correctly 1`] = `
<div
  className="cart-footer"
>
  <a
    className="continue-shopping"
    href="/"
    onClick={[Function]}
  >
    CONTINUE SHOPPING
  </a>
  <button
    className="cart-footer-button"
  >
    CHECKOUT ($
    0.00
    )
  </button>
</div>
`;
